docs/v1/P4Runtime-Spec.mdk

```diff
@@ -6304,6 +6324,380 @@ field [@ProtoAny]. At the moment, there is not any mechanism to extend the
+
+* `repeated string generic_table_properties`, please check generic-table
+  properties under section [GenericTable categories]
+  (#sec-generic-table-categories) for more details
+
+* `repeated GenericTableInstance`, contains the info for one instance of
+  a generic-table-type. All tables of one type are grouped under this.
+
+GenericTableInstance
+* `preamble`, a `Preamble` message with the ID, name, and alias of this
+  GenericTable.
+
+* `generic_match_fields`, a repeated field of type `GenericMatchField`
+  representing the data to be used to construct the lookup key matched in this
+  table. For information check the generic_match_fields info in the [Generic
+  Table section](#sec-generic-table-entry)
+
+* `union_refs`, a repeated `UnionRef` field representing the set of possible
+  unions for this table. Functionally, it behaves same as that of ActionRefs.
+  Hence it has been kept as ActionRefs itself.
+  Please check the union_refs info in the [Generic Table section]
+  (#sec-generic-table-entry)
+
+* `const_default_union_id`, if this table has a constant default union, this
+  field will carry the `uint32` identifier of that action, otherwise its value
+  will be 0. A default union is the data when the key is null. It is similar to
+  a default action for Match Tables. Being constant means that the control
+  plane cannot set a different default union at runtime or change the default
+  union's arguments.
+
+* `size`, an `int64` describing the desired number of table entries that the
+  target should support for the table.  See the "Size" subsection within the
+  "Table Properties" section of the P4~16~ language specification for details
+  [@P4TableProperties].
+
+* `other_properties`, an `Any` Protobuf message [@ProtoAny] to embed
+  architecture-specific or target-specific table properties [@P4TableProperties]
+  that the target wants to convey.
+
+## `GenericData` { #sec-p4data-generic-data}
+
+`GenericData` is an addition to p4data.proto- which aims at adding support for
+data types which aren't covered by P4Data. This incorporates all the data types
+used as primitive data types like bytes, float, strings. It contains p4type as
+well. P4type is to be used for any data type present in P4. If there is a
+conflict, like bool, then the P4 data type should be used.
+
+  * bitstring : Bytes type used for simple unsigned integers or bytes
+  * float : Floating type values. Target might return a lower-precision
+    ceiled value even if field is non-volatile. Target might choose to show a
+    field as volatile if it allows user to insert a certain value but does its
+    own modifications to the value when writing to hardware.
+  * bool : Boolean type values
+  * generic_struct : Used to define a structure that can contain
+    one or more of the other members as named. One structure can contain
+    different members of different types in GenericData.
+  * generic_list : Used to define a list of same types. Only allows for a
```

Comment:
"a list of same types" -> "a list of values, all with the same type as each other"